grading/load_config_json.cpp

```diff
@@ -1404,3 +1419,28 @@ void VerifyGraderDeductions(nlohmann::json &json_graders) {
     std::cout << "ERROR! DEDUCTION SUM < 1.0: " << sum << std::endl;
   }
 }
+
+std::vector<std::string> gatherAllTestcaseIds(const nlohmann::json& complete_config){
+  std::vector<std::string> testcase_names;
+  nlohmann::json::const_iterator testcases = complete_config.find("testcases");
+  assert (testcases != complete_config.end());
+  for (nlohmann::json::const_iterator tc = testcases->begin(); tc != testcases->end(); tc++) {
+    std::string testcase_id = tc->value("testcase_id", "");
+    assert(testcase_id != "");
+    testcase_names.push_back(testcase_id);
+  }
+
+  nlohmann::json::const_iterator item_pool = complete_config.find("item_pool");
+  if (item_pool != complete_config.end()){
+    for(nlohmann::json::const_iterator item_itr = item_pool->begin(); item_itr != item_pool->end(); item_itr++) {
+      nlohmann::json item_testcases = item_itr->value("testcases", nlohmann::json::array());
+      for(nlohmann::json::iterator it_itr = item_testcases.begin(); it_itr != item_testcases.end(); it_itr++) {
+        std::string testcase_id = it_itr->value("testcase_id", "");
+        assert(testcase_id != "");
+        testcase_names.push_back(testcase_id);
+      }
+    }
+  }
+
+  return testcase_names;
```

Comment:
Might be good to throw an assertion in here that this return value vector has no duplicate values.